### DISCUSSION TITLE
Using --install-types for older packages

### DISCUSSION THREAD

**Discussion by jenstroeger** [python/typing#1268] [General]
Hi,

I’m playing around with the [`--install-types`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-install-types) option (or `install_types = true` in a configuration file) and expected it to download & install stub packages for older packages too.

For example, I’ve got `jsonschema ==4.14.0` installed and I can manually install `types_jsonschema ==4.14.0` and everything works fine. However, using the `--install-types` option fails in this case because, I suspect, its [METADATA.toml](https://github.com/python/typeshed/blob/6950c3237065e6e2a9b64810765fec716252d52a/stubs/jsonschema/METADATA.toml) declares `version = "4.16.*"`. 

Without continuously updating my package to meet Typeshed’s type stubs version, how can I have `mypy` install the correct stubs?

Thanks!

**Comment by hauntsaninja** [comment]
TBH, I don't really recommend using mypy's `--install-types`. Just use a requirements.txt file listing the `types-*` packages you need and install them before running mypy.
